pages/agent/clusters.md

```diff
@@ -0,0 +1,97 @@
+
+The following diagram shows the architectural change when enabling clusters.
+
+<%= image "clusters-architecture.png", alt: "Diagram showing existing architecture and architecture with clusters" %>
+
+Clusters encapsulate groups of agents and pipelines. This enables the following:
+
+* Clusters are viewable to your entire organization, allowing engineers to better understand the agents and queues available for their pipelines.
+* Individual users or teams can maintain their own clusters. Cluster maintainers can manage queues and agent tokens and add and remove pipelines.
+* Pipelines can be assigned to a cluster, ensuring their builds run only on the agents connected to this cluster. These pipelines can also trigger builds only on other pipelines in the same cluster.
+
+## Enabling clusters
+
+Any Buildkite administrator can enable clusters for an organization. Once you enable clusters, you can only disable it by contacting support.
+
+To enable clusters:
+
+1. Navigate to your [organization’s pipeline settings](https://buildkite.com/organizations/~/pipeline-settings).
+1. In _Clusters_, select _Enable Clusters_.
+
+_Clusters_ will now appear in the global navigation.
+
+## Setting up a cluster
+
+To set up a new cluster:
+
+1. Navigate to _Clusters_
+2. Click _Create a Cluster_
+3. Enter a name, description and emoji
+4. Click _Create Cluster_
+
+### Setting up queues
+
+When you create your first cluster, it will have an initial “default” queue.
+
+To create additional queues:
+
+1. Navigate to the cluster’s _Queues_
+2. Click _Create a Queue_
+3. Enter a key and description
+4. Click _Create Queue_
+
+You should [supply the queue’s key][agent-queue-key] to the agents that you wish to run build jobs on the queue.
+
+[agent-queue-key]: /docs/agent/v3/queues#setting-an-agents-queue
+
+### Using cluster agent tokens
+
+Once you’ve created your cluster and queues, navigate to the cluster’s _Agent tokens_, copy the auto-generated token and [use this token][agent-use-token] for the agents you wish to connect to the cluster.
+
+You can also create, edit, and revoke other agent tokens here.
+
+[agent-use-token]: /docs/agent/v3/tokens#using-and-storing-tokens
+
+### Adding pipelines to a cluster
+
+Add a pipeline to a cluster to ensure the pipeline’s builds run only on agents connected to that cluster.
+
+To add a pipeline to a cluster:
+
+1. Navigate to the _Pipeline Settings_ for the pipeline.
+1. Under _Cluster Settings_, select the relevant cluster.
+
+### Adding maintainers to a cluster
+
+Only Buildkite administrators or users with the [“change organization” permission][permissions] can create clusters.
+
+Afterwards, you can assign other users or teams as a cluster’s maintainers to permit them to manage the cluster. Cluster maintainers can:
```

Comment:
After what?

```suggestion
You can assign other users or teams as a cluster’s maintainers to permit them to manage the cluster. Cluster maintainers can:
```

Reply:
The "after" here is after creating the cluster. But I can see how this is redundant.

That said, there is definitely a _sequence_ that I was hoping to imply here:

1. You need to be an _admin_ user to create the cluster in the first place
2. After creating the cluster, you can assign maintainers so that _non-admin_ users can manage it

If I could make one tweak to your suggestion, I would possibly make it:

> You can **then** assign...

(emphasis to show the added word)

What do you reckon? Any other possible approaches to emphasise how assigning cluster maintainers allows otherwise unprivileged users to be able to manage a cluster?

Reply:
"After" here is after creating the cluster, but I can see how that's not necessary. Adjusted!

Reply:
Ahhhh okay, I don't think that ordering is super clear. Hmmm. Maybe:

> After a cluster is created, you can ...

I also wonder whether this section should be more general to "Cluster permissions" rather than just maintainers. 🤔